security/providers/oidc-common/src/main/java/io/helidon/security/providers/oidc/common/spi/TenantIdProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.security.providers.oidc.common.spi;
+
+import io.helidon.config.Config;
+
+/**
+ * Java {@link java.util.ServiceLoader} service interface for multitenancy support.
+ */
+public interface TenantIdProvider {
```

Comment:
why do we have two/three levels of abstraction here? Provider -> Finder -> Optional<String> ?

1. I can see the need for Provider finding the 1st Finder that is capable of returning a non-optional tenant-id, in weighted order.  Is that the intent?  If so then we should provide methods on Provider that do sort of thing.
2. If that is not the intent then why not just use serviceloader directly on the Finder; no need to go to provider first.

Reply:
With having TenantIdProvider it is possible to create new TenantIdFinder with the config. (user can specify their own config structure how to create finder)
Finder itself is there for finding the tenant id and if it is missing (tenant id), we do not want to return null, but rather Optional. That is the reason for structure like this.